docs/benchmarks-design.md

```diff
@@ -0,0 +1,324 @@
+# <!-- omit in toc --> Superbenchmark Benchmarks Abstraction
+
+## <!-- omit in toc -->Table of Content
+- [Introduction](#introduction)
+  - [Overview](#overview)
+  - [Goals](#goals)
+- [High-level Technical Design](#high-level-technical-design)
+- [Detailed Technical Design](#detailed-technical-design)
+  - [ModelBase Benchmarks](#modelbase-benchmarks)
+    - [Training](#training)
+    - [Inference](#inference)
+  - [Micro Benchmarks](#micro-benchmarks)
+  - [DockerBase Benchmarks](#dockerbase-benchmarks)
+  - [Benchmark Registry](#benchmark-registry)
+    - [Design](#design)
+    - [Examples](#examples)
+- [Interfaces](#interfaces)
+  - [Inputs](#inputs)
+  - [Invoke](#invoke)
+  - [Outputs](#outputs)
+    - [Design](#design-1)
+    - [Example](#example)
+- [Appendix](#appendix)
+
+## Introduction
+
+### Overview
+
+Superbenchark is a benchmark framework that contains lots of useful micro-benchmarks and E2E-benchmarks targeting for hardware/software validation,  performance measurement and offering advice for better resource utilization. For now, Superbenchmark suppert:
+
+Micro-benchmark: FIO, P2P_Bandwidth, TensorCore_Performance, GPU_FLOPS, TCP_Performance, RDMA_Performance, MPI_Performance, NCCL_Performance, Stress_Test.
+
+E2E-benchmark: 32-CNN models, LSTM, BERT, Seq2Seq, DeepSpeech2, NASNet, GraphSage
+
+However the current structure of all the benchmarks are very loose. The pain points we have met are as follows:
+
+1.	Every benchmark has an independent python file and there exist a lot of redundant/dead code. 
+2.	To launch one benchmark, we must specify the file path of the benchmark which is hard coded and pieced together with the test name string from config file. 
+3.	To add new micro-benchmarks for new accelerators, we must modify the related source code and add IF statement to distinguish different platforms.
+4.	To modify some common logic of E2E-benchmarks like Dataset, DataLoader and distributed related code, we need change all the E2E benchmark files.
+5.	It’s hard to extended new features, such as running benchmark for N times.
+6.	The output of each benchmark is disorganized and unstructured.
+7.	Some benchmarks have big variance, it’s hard to do the confirmation and investigation manually.
+8.	Hard to detect the interruption of running, especially for microbenchmarks which is launched by system call.
+
+This document will present a new design for the benchmarks to avoid the above pain points. 
```

Comment:
Therefore, we would like to re-architecture Benchmarks abstraction in SuperBench v0.2 to avoid all pain points mentioned above.